.github/workflows/prerelease-head.yml

```diff
@@ -0,0 +1,132 @@
+      env: |
+        echo "GHC_FOR_RELEASE=${{ steps.conf.outputs['GHC_FOR_RELEASE'] }}" >> "$GITHUB_ENV"
+        echo "GHC_FOR_SOLVER_BENCHMARKS=${{ steps.conf.outputs['GHC_FOR_SOLVER_BENCHMARKS'] }}" >> "$GITHUB_ENV"
+        echo "GHC_FOR_COMPLETE_HACKAGE_TESTS=${{ steps.conf.outputs['GHC_FOR_COMPLETE_HACKAGE_TESTS'] }}" >> "$GITHUB_ENV"
+        echo "GHC_FOR_VALIDATE=${{ steps.conf.outputs['GHC_FOR_VALIDATE'] }}" >> "$GITHUB_ENV"
+        echo "GHC_FOR_VALIDATE_OLD=${{ steps.conf.outputs['GHC_FOR_VALIDATE_OLD'] }}" >> "$GITHUB_ENV"
+        echo "COMMON_FLAGS=${{ steps.conf.outputs['COMMON_FLAGS'] }}" >> "$GITHUB_ENV"
+        echo "LTS_RELEASE=${{ steps.conf.outputs['LTS_RELEASE'] }}" >> "$GITHUB_ENV"
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: pietrobolcato/action-read-yaml@1.1.0
+        id: conf
+        with:
+          config: ${{ github.workspace }}/.github/config.yml
+
+  prerelease-build-most:
+    name: Prerelease build ${{ matrix.sys.os }} ghc-${{ needs.config.outputs.GHC_FOR_RELEASE }}
+    runs-on: ${{ matrix.sys.os }}
+    needs: config
+    strategy:
+      fail-fast: false
+      matrix:
+        sys:
+          # 'bash' on Windows apparently gets you the one from Git for Windows,
+          # whereas this needs the one from msys
+          - { os: windows-latest, shell: "C:/msys64/usr/bin/bash.exe -e {0}" }
+          - { os: ubuntu-22.04, shell: bash }
+          - { os: macos-latest, shell: bash }
+          - { os: macos-13, shell: bash }
+    defaults:
+      run:
+        shell: ${{ matrix.sys.shell }}
+
+    steps:
+      - run: ${{ needs.config.outputs.env }}
+
+      - uses: actions/checkout@v4
+
+      # We were making prereleases with validate builds, so I will continue to do so
+      - uses: ./.github/actions/validate-build
+        with:
+          shell: ${{ matrix.sys.shell }}
+          ghc: ${{ needs.config.outputs.GHC_FOR_RELEASE }}
+
+  prerelease-build-alpine:
+    name: Prerelease build statically linked using alpine
+    runs-on: ubuntu-latest
+    container: "alpine:3.19"
+    needs: config
+    steps:
+      - run: ${{ needs.config.outputs.env }}
+        shell: sh
+
+      - name: Install extra dependencies
+        shell: sh
+        run: |
+          apk add bash curl sudo jq pkgconfig \
+          zlib-dev zlib-static binutils-gold curl \
+          gcc g++ gmp-dev libc-dev libffi-dev make \
+          musl-dev ncurses-dev perl tar xz
+
+      - uses: actions/checkout@v4
+
+      - uses: ./.github/actions/validate-build
+        with:
+          shell: 'bash'
+          ghc: ${{ needs.config.outputs.GHC_FOR_RELEASE }}
+          static: 'true'
+
+  prerelease-head:
+    name: Create a GitHub prerelease with the binary artifacts
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+
+    needs: [prerelease-build-most, prerelease-build-alpine]
+
+    steps:
+      # for now this is hardcoded. is there a better way?
```

Comment:
You could use a keyword like NOTE, TODO or HACK so that automated collection of such notes can pick it up.

Reply:
I think the referent for that comment went away: it came from `validate.yml` unchanged, but when we switched release actions we started using a wildcard instead of explicitly listing files and the comment was removed from `validate.yml` but not the prerelease jobs here.